File: 백준/Silver/1475. 방 번호/방 번호.py
num = int(input())
deciNum = 0
deci = [0] * 9
while num > 0:
    deciNum = num % 10
    if deciNum == 9:
        deciNum = 6
    deci[deciNum] += 1
    num //= 10

deci[6] = deci[6] // 2 if deci[6] % 2 == 0 else deci[6] // 2 + 1

max = 0
for check in deci:
    if max < check:
        max = check

print(max)

File: 백준/Silver/1475. 방 번호/방 번호.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(System.in);
        int num = sc.nextInt();
        sc.close();
        int[] deci = new int[9];
        while (num > 0) {
            int deciNum = num % 10;
            if (deciNum == 9) {
                deciNum = 6;
            }
            deci[deciNum]++;
            num /= 10;
        }
        deci[6] = deci[6] % 2 == 0 ? deci[6] / 2 : deci[6] / 2 + 1;

        int max = 0;
        for (int check : deci) {
            if (max < check)
                max = check;
        }
        System.out.println(max);
    }
}